dataplane/src/packet_processor/ipforward.rs

```diff
@@ -66,7 +66,7 @@ impl IpForwarder {
 
         /* get destination ip address */
         let Some(dst) = packet.ip_destination() else {
-            error!("{nfi}: Failed to get destination ip address for packet");
+            debug!("{nfi}: Failed to get destination ip address for packet");
```

Comment:
I see the intent here (reducing verbosity), but to me the changes in this commit will harm more than anything.
If something is an ERROR or a warning, we should log it as such, because when displayed the severity/loglevel tells you what it is. We now have the option to completely disable traces in the pipeline. We may want to adjust the levels in the source code if some errors get very frequently logged; and we still have the option to rate-limit them for instance. However, in this particular example, I'd say you'd want to see the log as an ERROR in neon-lights since failing to get the destination ip address should be the symptom of something going really wrong.

Reply:
If this can be triggered by a carefully crafted user packet, then this should *not* be error level as then malicious senders can cause logs to fill up and DOS the gateway.

Reply:
if the user can drive us into any situation where ip address is missing then this message is a dos vector

Reply:
Well, those packets (a packet with ethertype=IP and no IP header) should not even make it here, right?

Reply:
sure, but `error!` is intended for serious application level errors.  This is just what is happening if we happen to get invalid data.

Or are you concerned that we might manipulate the data into an invalid shape by accident (which, I suppose, would promote this to `error!` in truth)

Reply:
As discussed, I believe that to address your concerns (DoS) we should clearly delimit where it is safe to log "issues" and where not. In my mind, if I have an Ip packet and can't retrieve the ip address:
  * either there is a bug in the getter (unlikely), and that should be error
  * or the packet is malformed, in which case it should not be a valid packet and make it here (this is the IP-forward stage).
  
I agree that the parser may not know up to which "layer" it should validate a packet. But we should decide at which point packets can be considered malformed/well-formed so that we can "reasonably" tag logs with the right severity and at the same time safely be protected against DoS.

Reply:
fair enough.  It is left at error for now